.scripts/util.sh

```diff
@@ -38,85 +38,88 @@ copy_modules() {
-	cp "$SRC/sys/xdd/mfp/mfp.xdd" "$MCHDIR"
+	cp "$SRC/sys/sockets/xif/lance_000.xif" "$MCHDIR/lance.xix"
+	cp "$SRC/sys/sockets/xif/rieblmst_000.xif" "$MCHDIR/riebl.xix"
+	cp "$SRC/sys/sockets/xif/rieblmst_fast_000.xif" "$MCHDIR/riebl_fast.xix"
+	cp "$SRC/sys/sockets/xif/rieblspc_000.xif" "$MCHDIR/rieblspc.xix"
+	cp "$SRC/sys/sockets/xif/rieblspc_fast_000.xif" "$MCHDIR/riebls_fast.xix"
+	cp "$SRC/sys/sockets/xif/rtl8012st_000.xif" "$MCHDIR/rtl8012st.xix"
+	cp "$SRC/sys/xdd/mfp/.compile_000/mfp.xdd" "$MCHDIR"
 }
 copy_ste_modules() {
 	local MCHDIR="$1/ste"
 	mkdir -p "$MCHDIR"
-	cp "$SRC/sys/sockets/xif/rtl8012st.xif" "$MCHDIR/rtl8012st.xix"
-	cp "$SRC/sys/xdd/mfp/mfp.xdd" "$MCHDIR"
+	cp "$SRC/sys/sockets/xif/rtl8012st_000.xif" "$MCHDIR/rtl8012st.xix"
+	cp "$SRC/sys/xdd/mfp/.compile_000/mfp.xdd" "$MCHDIR"
 }
 copy_megaste_modules() {
 	local MCHDIR="$1/megaste"
 	mkdir -p "$MCHDIR"
-	cp "$SRC/sys/sockets/xif/rieblste.xif" "$MCHDIR/rieblste.xix"
-	cp "$SRC/sys/sockets/xif/rtl8012st.xif" "$MCHDIR/rtl8012st.xix"
-	cp "$SRC/sys/xdd/mfp/mfp.xdd" "$MCHDIR"
-	cp "$SRC/sys/xdd/scc/scc.xdd" "$MCHDIR"
+	cp "$SRC/sys/sockets/xif/rieblste_000.xif" "$MCHDIR/rieblste.xix"
+	cp "$SRC/sys/sockets/xif/rtl8012st_000.xif" "$MCHDIR/rtl8012st.xix"
+	cp "$SRC/sys/xdd/mfp/.compile_000/mfp.xdd" "$MCHDIR"
+	cp "$SRC/sys/xdd/scc/.compile_000/scc.xdd" "$MCHDIR"
 }
 copy_tt_modules() {
 	local MCHDIR="$1/tt"
+	local TARGET="$2"
 	mkdir -p "$MCHDIR"
-	cp "$SRC/sys/sockets/xif/daynaport/scsilink.xif" "$MCHDIR/scsilink.xix"
-	cp "$SRC/sys/sockets/xif/lance.xif" "$MCHDIR/lance.xix"
-	cp "$SRC/sys/sockets/xif/riebltt.xif" "$MCHDIR/riebltt.xix"
-	cp "$SRC/sys/sockets/xif/rtl8012.xif" "$MCHDIR/rtl8012.xix"
-	cp "$SRC/sys/xdd/mfp/mfp.xdd" "$MCHDIR"
-	cp "$SRC/sys/xdd/scc/scc.xdd" "$MCHDIR"
+	cp "$SRC/sys/sockets/xif/daynaport/.compile_030/scsilink.xif" "$MCHDIR/scsilink.xix"
+	cp "$SRC/sys/sockets/xif/lance_030.xif" "$MCHDIR/lance.xix"
+	cp "$SRC/sys/sockets/xif/riebltt_030.xif" "$MCHDIR/riebltt.xix"
+	cp "$SRC/sys/sockets/xif/rtl8012_030.xif" "$MCHDIR/rtl8012.xix"
+	cp "$SRC/sys/xdd/mfp/.compile_030/mfp.xdd" "$MCHDIR"
+	cp "$SRC/sys/xdd/scc/.compile_030/scc.xdd" "$MCHDIR"
 }
 copy_falcon_modules() {
 	local MCHDIR="$1/falcon"
 	mkdir -p "$MCHDIR"
-	cp "$SRC/sys/sockets/xif/rtl8012.xif" "$MCHDIR/rtl8012.xix"
-	cp "$SRC/sys/sockets/xif/daynaport/scsilink.xif" "$MCHDIR/scsilink.xix"
-	cp "$SRC/sys/xdd/dsp56k/dsp56k.xdd" "$MCHDIR"
-	cp "$SRC/sys/xdd/scc/scc.xdd" "$MCHDIR"
+	cp "$SRC/sys/sockets/xif/rtl8012_02060.xif" "$MCHDIR/rtl8012.xix"
+	cp "$SRC/sys/sockets/xif/daynaport/.compile_02060/scsilink.xif" "$MCHDIR/scsilink.xix"
+	cp "$SRC/sys/xdd/dsp56k/.compile_02060/dsp56k.xdd" "$MCHDIR"
+	cp "$SRC/sys/xdd/scc/.compile_02060/scc.xdd" "$MCHDIR"
 }
 copy_milan_modules() {
 	local MCHDIR="$1/milan"
 	mkdir -p "$MCHDIR"
-	cp "$SRC/sys/xdd/uart/uart.xdd" "$MCHDIR"
-	cp "$SRC/sys/xdd/mfp/mfp_mil.xdd" "$MCHDIR"
+	cp "$SRC/sys/xdd/uart/.compile_mil/uart.xdd" "$MCHDIR"
```

Comment:
See above, while mil = 02060 is good for the generic bootable archive, it could use 040 and 060 for the CPU builds.

Reply:
Well, the reason for existence of the CPU builds is itself very thin with the `sysdir.tos` and #32 being implemented (my initial idea was to provide them for a) stubborn users requiring their CPU be utilised b) users which just wish to extract&overwrite their existing installation).

So making every single module optimised for the given CPU seems like a nice feature. Otherwise we may really consider killing the whole "CPU builds" concept.